scanpy/_utils/__init__.py

```diff
@@ -857,6 +858,13 @@ def warn_with_traceback(message, category, filename, lineno, file=None, line=Non
     settings.write(warnings.formatwarning(message, category, filename, lineno, line))
 
 
+def warn_once(msg: str, category: type[Warning], stacklevel: int = 1):
+    warnings.warn(msg, category, stacklevel=stacklevel)
+    # Prevent from showing up every time an awkward array is used
+    # You'd think `'once'` works, but it doesn't at the repl and in notebooks
```

Comment:
What is this about awkward arrays?

Reply:
```suggestion
    # You'd think `'once'` works, but it doesn't at the repl and in notebooks
```

Cause this is copied from anndata